### PR TITLE
Create GitHub Actions workflow to build Linux and MacOS binaries and attach them to the latest release

### PR DESCRIPTION
**Pull request by jmoore34**
Usage: create a release and the workflow will be triggered and will build and upload the binaries

[Example execution](https://github.com/jmoore34/typocide/releases/tag/0.0.5)

### CODE REVIEW

File: .github/workflows/upload_binaries_to_release.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        with:
+          ocaml-compiler: "5.1"
+          opam-repositories: |
+            default: https://github.com/ocaml/opam-repository.git
+
+      - run: opam install . --deps-only
+
+      - run: opam exec -- dune build
+
+      - run: mv ./_build/default/bin/main.exe ./_build/default/bin/Typocide_${{ runner.os }}.exe
```
Comment (rdavison) [inline .github/workflows/upload_binaries_to_release.yml]: Can you change `Typocide_${{ runner.os }}.exe` to `typocide-${{ runner.os }}`. Eventually when we get a windows build, we can conditionally attach a `.exe` suffix to it for windows only.

File: .github/workflows/upload_binaries_to_release.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      
+      - run: ls -laR .
+        continue-on-error: true
+
+      - name: Release
+        uses: softprops/action-gh-release@v1
+        with:
+          files: |
+            ./_build/default/bin/Typocide_${{ runner.os }}.exe
```
Comment (rdavison) [inline .github/workflows/upload_binaries_to_release.yml]: Here too.

**Comment by rdavison** [comment]
This is really great, thank you for the contribution! It's a bummer about the windows build, but I can poke around and try to get it working later in a separate PR. For now, I think this is great. I left a comment about a minor change. Once that's done we can merge it.

File: .github/workflows/upload_binaries_to_release.yml (hunk 1 above)
Comment (jmoore34) [reply .github/workflows/upload_binaries_to_release.yml]: Okay, removed

File: .github/workflows/upload_binaries_to_release.yml (hunk 2 above)
Comment (jmoore34) [reply .github/workflows/upload_binaries_to_release.yml]: Updated

**Pull request merged**